File: hyuksanglee/01/[PGS - Lv2] H-Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        int end = citations.length-1;
        
        int index =1;
        for(int i = end; i>=0; i--){
            
           int num = citations[i];
            if(num<index){
                return index-1;
            }
                index++;
        }
        
       
        
        return citations.length;
    }
}

File: 6월 4주차/JeongEon8/01/[PGS - Lv2] H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>


using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.begin(), citations.end());
    
    int size = citations.size();

    for(int h = citations[size - 1]; h >= 0; h--){
        int cnt = 0;
        for(int j = 0; j < size; j++){
            if(citations[j] >= h){
                cnt++;
            }
        }
        if(cnt >= h && size - cnt <= h){
            answer = h;
            break;
        }
    }
    
    return answer;
}
